src/main/java/pageFactory/ProductPage.java

```diff
@@ -0,0 +1,152 @@
+package pageFactory;
+
+import base.BasePage;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProductPage extends BasePage {
+    @FindBy(css= ".nav-1")
+    private static WebElement whatsNewLink;
+
+    @FindBy(xpath= "(//*[@class=\"item\"])[position()=1]")
+    private static WebElement categoryLink;
+
+    @FindBy(xpath= "(//*[@class=\"product-image-photo\"])[position()=1]")
+    private static WebElement productLink;
+
+    @FindBy(xpath= "(//*[@class=\"action towishlist\"])[position()=1]")
+    private static WebElement addToWishButton;
+
+    @FindBy(css= ".toolbar-number")
+    private static WebElement wishlistItems;
+
+    @FindBy(xpath= "(//*[@class=\"swatch-option text\"])[position()=1]")
+    private static WebElement size;
+
+    @FindBy(xpath= "(//*[@class=\"swatch-option color\"])[position()=1]")
+    private static WebElement color;
+
+    @FindBy(css= "#product-addtocart-button")
+    private static WebElement addToCartButton;
+
+    @FindBy(css= ".counter-number")
+    private static WebElement cartCounter;
+
+    @FindBy(css= ".viewcart")
+    private static WebElement viewAndEditCartLink;
+
+    @FindBy(css= ".product-image-photo")
+    private static WebElement productWishlist;
+
+    @FindBy(css= ".btn-remove")
+    private static WebElement removeProductWishlistButton;
+
+    @FindBy(css= ".form-cart")
+    private static WebElement formCart;
+
+    @FindBy(xpath= "(//*[@class=\"action action-delete\"])[position()=1]")
+    private static WebElement product;
+
+
+
+    public ProductPage(WebDriver driver) throws MalformedURLException, URISyntaxException {
+        super(driver);
+        initElements();
+
+    }
+
+    public  ProductPage clickWhatsNewLink() throws MalformedURLException, URISyntaxException {
+        whatsNewLink.click();
+        return new  ProductPage(driver);
+    }
+
+    public  ProductPage clickCategoryLink() throws MalformedURLException, URISyntaxException {
+        categoryLink.click();
+        return new  ProductPage(driver);
+    }
+    public  ProductPage clickProductLink() throws MalformedURLException, URISyntaxException {
+        productLink.click();
+        return new  ProductPage(driver);
+    }
+
+    public  ProductPage addProduct() throws MalformedURLException, URISyntaxException {
```

Comment:
let's add more flexibility: I want to search by name of the product, names of the product should be passed as the parameter to this method